src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -402,6 +402,7 @@ private void LowerConstantValueDecision(DecisionTree.ByValue byValue)
                     {
                         var loweredRight = kv.Key;
                         var loweredLeft = kv.Value;
+                        loweredRight = _factory.Convert(loweredLeft.Type, loweredRight);
```

Comment:
The tests cover many kinds of conversions. The newly added tests fill in a gap in the kinds of conversions that had been tested before.